components/rmrk/Create/CreateCollection.vue

```diff
@@ -189,43 +131,37 @@ export default class CreateCollection extends mixins(
     this.isLoading = true
     this.status = 'loader.ipfs'
 
+    const { transaction, status, isLoading, blockNumber } = useTransaction()
+
+    watch([isLoading, status], () => {
+      this.isLoading = isLoading.value
+      if (Boolean(status.value)) {
+        this.status = status.value
+      }
+    })
+    watch(blockNumber, (block) => {
+      if (block) {
+        this.$emit('created')
+      }
+    })
+    showNotification(
+      `Creating collection ${this.base.name}`,
+      notificationTypes.info
+    )
+
     try {
-      const metadata = await this.constructMeta()
-      const mint = this.constructRmrkMint(metadata)
-      const mintInteraction = createMintInteaction(
-        Interaction.MINT,
-        this.version,
-        mint
-      )
-
-      const api = await this.useApi()
-      showNotification(
-        `Creating collection ${this.base.name}`,
-        notificationTypes.info
-      )
-
-      const cb = !this.hasSupport
-        ? api.tx.system.remark
-        : api.tx.utility.batchAll
-      const args = !this.hasSupport
-        ? mintInteraction
-        : [
-            asSystemRemark(api, mintInteraction),
-            ...(await canSupport(api, this.hasSupport)),
-          ]
-
-      await this.howAboutToExecute(
-        this.accountId,
-        cb,
-        [args],
-        (blockNumber) => {
-          showNotification(
-            `[Collection] Saved ${this.base.name} in block ${blockNumber}`,
-            notificationTypes.success
-          )
-          this.$emit('created')
-        }
-      )
+      const { description, file, name } = this.base
+
+      transaction({
+        interaction: Interaction.MINT,
+        urlPrefix: usePrefix().urlPrefix.value,
+        description,
+        file,
+        name,
+        tags: [],
+        count: this.unlimited ? 0 : this.max,
+        symbol: this.symbol,
```

Comment:
Isn't better to wrap it into the other payload object?

Reply:
agree
I've changed the signature to 
```ts
export interface ActionMintCollection {
  interaction: Interaction.MINT
  urlPrefix: string
  collection: CollectionToMintBasilisk | CollectionToMintKusama
  successMessage?: string | ((blockNumber: string) => string)
  errorMessage?: string
}
```